apis/api_test.go

```diff
@@ -0,0 +1,103 @@
+package apis_test
+
+import (
+	"cloudfoundry.org/cf-k8s-api/apis"
+	. "github.com/onsi/gomega"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+const (
+	jsonHeader = "application/json"
+	defaultServerURL = "https://api.example.org"
+)
+func TestAPI(t *testing.T) {
+	spec.Run(t, "object", testRootV3API, spec.Report(report.Terminal{}))
+	spec.Run(t, "object", testRootAPI, spec.Report(report.Terminal{}))
+}
+
+func testRootV3API(t *testing.T, when spec.G, it spec.S) {
+	Expect := NewWithT(t).Expect
+
+	when("the v3 GET endpoint returns successfully", func() {
+		// Create a request to pass to our handler. We don't have any query parameters for now, so we'll
+		// pass 'nil' as the third parameter.
+		req, err := http.NewRequest("GET", "/v3", nil)
```

Comment:
I'm still not 100% clear on what's safe in spec, but having this shared setup outside of an `it.Before` may lead to test pollution or other problems in the future.

Reply:
Agreed. Couldn't find documentation explaining what the default behavior for code inside `when()` is. As `it.Before` is well defined and documented, enclosing within `it.Before` is good.  

Change done as part of commit - [ec3de58a915baa698d7c229c2dccb67f4a9b8c05](https://github.com/cloudfoundry/cf-k8s-api/pull/5/commits/ec3de58a915baa698d7c229c2dccb67f4a9b8c05)